### ISSUE TITLE
Empty pandas timestamp (NaT) throws error when engine converts to pandas.Series.dt.strftime

### ISSUE THREAD

**Issue by ivandochev** [trinodb/trino-python-client#142]
Hi guys,

Noticed that while None and pd.nan are possible parameters for the execute method, if there is a pandas NaT value somewhere in the parameters, this throws an error when the cursor tries to convert it to formatted time string. The workaround is to do sth like:

params = [val if type(val) != pd._libs.tslibs.nattype.NaTType else None
                            for row in temp_df.itertuples() for val in row[1:]]

but it isn't pretty.

Since other empty types are supported with no problem, it seems a good idea to have a NaT check before trying to convert the timestamp with .dt.strftime and return None if the value is NaT.

Cheers

**Comment by BeRT2me** [comment]
Why not preempt the issue with something like this?
```df.replace(pd.NaT, None, inplace=True)```

*Requires pandas version 1.4.0+

**Comment by hashhar** [comment]
The fact that `pd.NaN` works at all is just a co-incidence. The parameters are not type-safe yet. They just whatever `__str__` returns IIRC. Follow https://github.com/trinodb/trino-python-client/issues/45 for the improvement. (There is no one actively working on it since it hasn't been a frequent enough issue to spend time on yet).

**Comment by mdesmet** [comment]
The issue here is that `pd.NaT` presents itself as a `datetime.datetime` while it doesn't allow `strftime` to be called upon itself. It's not an issue of type safety. 

You can perfectly insert it using `df.to_sql()` and sqlalchemy.

```
>>> import pandas as pd
>>> import datetime
>>> from sqlalchemy import create_engine
>>> isinstance(pd.NaT, datetime.datetime)
True
>>> engine = create_engine('trino://user@localhost:8080/memory/default')
>>> connection = engine.connect()
>>> data = {'col1': [pd.NaT]}
>>> df = pd.DataFrame(data)
>>> pd.api.types.infer_dtype(df['col1'])
'datetime64'
>>> df.to_sql("test_sqlalchemy",    con=connection,    schema="default",    if_exists="replace",    index=False)
-1
```

@hashhar: I don't think we should check for pandas types (if `param is pd.NaT`) within trino-python-client. My recommendation is to close this issue as won't do.

**Comment by hashhar** [comment]
I agree we shouldn't special case pandas. What I meant instead was that the client at the time didn't actually perform any explicit conversion from Python object to the string that gets added to the query - it used to just rely on whatever `__str__` for an object happened to return.

This was fixed in https://github.com/trinodb/trino-python-client/pull/51 though so I'll close this as won't do. And also close #45 since it was fixed in #51.
